itests/hive-unit/src/test/java/org/apache/hive/service/server/InformationSchemaWithPrivilegeTestBase.java

```diff
@@ -287,7 +291,7 @@ public void test() throws Exception {
 
     List<String> args = new ArrayList<String>(baseArgs);
     args.add("-f");
-    args.add("../../metastore/scripts/upgrade/hive/hive-schema-4.0.0-alpha-1.hive.sql");
+    args.add("../../metastore/scripts/upgrade/hive/hive-schema-" + hiveSchemaVer + ".hive.sql");
```

Comment:
Can we use `HiveVersionInfo.getShortVersion()` in these places too?

Reply:
I think that the  `HiveVersionInfo.getShortVersion()` and the `MetaStoreSchemaInfoFactory.get(miniHS2.getServerConf()).getHiveSchemaVersion()` might and should be different in several deployments. Some random vendor (😄) might release different code version (hotfixes, maintenance releases) for the same HMS DB version.

I think this is why `MetaStoreSchemaInfoFactory` has been created, and I tried to use the appropriate version in the appropriate place.

Reply:
Oh I see. Then I think you did it correctly

Reply:
So far we have:
- `HiveVersionInfo`
- `MetastoreVersionInfo`
- `MetaStoreSchemaInfo`
I understand there is a concern between HS2 and HMS version being different but what about `MetastoreVersionInfo`
vs `MetaStoreSchemaInfo`?

Reply:
* `HiveVersionInfo` is the code version of the HS2
* `MetastoreVersionInfo` is the code version of the HMS
* `MetaStoreSchemaInfo` is the db schema version under the HMS - some HMS changes do not require schema changes

Reply:
Thanks both for the clarifications!